File: app/src/main/java/linusfessler/alarmtiles/ViewPagerLinearBackgroundColorInterpolator.java
package linusfessler.alarmtiles;

import androidx.viewpager.widget.ViewPager;

public class ViewPagerLinearBackgroundColorInterpolator implements ViewPager.OnPageChangeListener {

    private final ViewPager viewPager;
    private final LinearColorInterpolator linearColorInterpolator;

    public ViewPagerLinearBackgroundColorInterpolator(final ViewPager viewPager, final int... colors) {
        this.viewPager = viewPager;
        linearColorInterpolator = new LinearColorInterpolator(colors);
    }

    @Override
    public void onPageScrolled(final int position, final float positionOffset, final int positionOffsetPixels) {
        final int backgroundColor = linearColorInterpolator.getColor(position + positionOffset);
        viewPager.setBackgroundColor(backgroundColor);
    }

    @Override
    public void onPageSelected(final int position) {
        // Nothing to do
    }

    @Override
    public void onPageScrollStateChanged(final int state) {
        // Nothing to do
    }
}

File: app/src/main/java/linusfessler/alarmtiles/ViewPagerLinearBackgroundColorInterpolator.kt
package linusfessler.alarmtiles

import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener

class ViewPagerLinearBackgroundColorInterpolator(private val viewPager: ViewPager, vararg colors: Int) : OnPageChangeListener {
    private val linearColorInterpolator: LinearColorInterpolator = LinearColorInterpolator(*colors)

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        val backgroundColor = linearColorInterpolator.getColor(position + positionOffset)
        viewPager.setBackgroundColor(backgroundColor)
    }

    override fun onPageSelected(position: Int) {
        // Nothing to do
    }

    override fun onPageScrollStateChanged(state: Int) {
        // Nothing to do
    }
}